api/candlepin-api-spec.yaml

```diff
@@ -3992,6 +3992,196 @@ components:
             serial:
               $ref: '#/components/schemas/CertificateSerialDTO'
 
+    CapabilityDTO:
+      description: Represents a consumer capability
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+
+    ConsumerDTO:
+      description: Represents a consumer
+      allOf:
+        - $ref: '#/components/schemas/TimestampedEntity'
+        - type: object
+          properties:
+              id:
+                type: string
+              uuid:
+                type: string
+              name:
+                type: string
+              username:
+                type: string
+              entitlementStatus:
+                type: string
+              serviceLevel:
+                type: string
+              role:
+                type: string
+              usage:
+                type: string
+              addOns:
+                type: array
+                uniqueItems: true
+                items:
+                  type: string
+              systemPurposeStatus:
+                type: string
+              releaseVer:
+                $ref: '#/components/schemas/ReleaseVerDTO'
+              owner:
+                $ref: '#/components/schemas/NestedOwnerDTO'
+              environment:
+                $ref: '#/components/schemas/EnvironmentDTO'
+              entitlementCount:
+                type: integer
+                format: int64
+              facts:
+                type: object
+                additionalProperties:
+                  type: string
+              lastCheckin:
+                type: string
+                format: date-time
+              installedProducts:
+                type: array
+                uniqueItems: true
+                items:
+                  $ref: '#/components/schemas/ConsumerInstalledProductDTO'
+              canActivate:
+                type: boolean
+              capabilities:
+                type: array
+                uniqueItems: true
+                items:
+                  $ref: '#/components/schemas/CapabilityDTO'
+              hypervisorId:
+                $ref: '#/components/schemas/HypervisorIdDTO'
+              contentTags:
+                type: array
+                uniqueItems: true
+                items:
+                  type: string
+              autoheal:
+                type: boolean
+              annotations:
+                type: string
+              contentAccessMode:
+                type: string
+              type:
+                $ref: '#/components/schemas/ConsumerTypeDTO'
+              idCert:
+                allOf:
+                  - $ref: '#/components/schemas/CertificateDTO'
+              guestIds:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GuestIdDTO'
```

Comment:
I believe, since this GuestIdDTO is within a list, we should instead use the GuestIdDTOArrayElement here.
I am not 100% sure though, so before doing that, you could check what current master does (e.g. does a GET /consumer/{consumer_uuid} for a consumer that has guest ids that themselves have attributes return the attributes too?)

Reply:
We are setting empty guest details for every `Consumer` translation to `ConsumerDTO`. https://github.com/candlepin/candlepin/blob/3c74b22ea291221b8c33da0bd9a1ab4197d956ad/server/src/main/java/org/candlepin/dto/api/v1/ConsumerTranslator.java#L176
Hence there is no need to update the `GuestIdDTOArrayElement` here.